src/originalBudgetTrackingApp/commonTypes.ts

```diff
@@ -1,7 +1,31 @@
 import { Transaction } from '@/originalBudgetTrackingApp/bankScraper';
+import { CompanyTypes } from '@brafdlog/israeli-bank-scrapers-core';
+
+export interface ProgressEmitter {
+  onProgress(func: (...args: any[]) => void): void
+}
 
 export interface EnrichedTransaction extends Transaction {
   accountNumber: string;
   category?: string;
   hash: string;
 }
+
+export interface ScrapingEvents {
+  status: (status: string, data?: { name?: string, [other: string]: any }) => void,
+  progress: (...args: any[]) => void,
+  error: (message: string, data?: { companyId?: CompanyTypes, error?: Error, [other: string]: any }) => void,
+  finish: (data?: { companyId?: CompanyTypes, accountId?: string, [other: string]: any }) => void
+}
+
+export interface ScrapingEventEmitter {
```

Comment:
I don't require a "real" event emitter (`require('events').EventEmitter`) because of `EventEmitter` is just an implementation of the `emit` method.
Actually now I think this interface even should not include the `on` method, because you need to decide if you really need this method. The scraping method doesn't use this method.
(See in the `ConsoleEmitter`, you don't have to use this method at all)